### DISCUSSION TITLE
REXX levelb assembler instruction

### DISCUSSION THREAD

**Discussion by Peter-Jacob** [adesutherland/CREXX#138] [Q&A]
due to some missing do loop options, I wanted to outbreak a loop using the assembler instruction BR.
e.g.    ` if pos(needle,haystack,offset)=0 then assembler br c900`

fulls sample:
```
/* rexx */
options levelb
...
/* countstr(needle,haystack) returns number occurrences of needle in haystack */
countstr: procedure = .int
  arg expose needle = .string, expose haystack = .string /* Pass by reference */
  count=0
  offset=1
  C100:
     if pos(needle,haystack,offset)=0 then assembler br c900
     count=count+1
     offset=offset+1
     assembler br C100
  C900:
return count
```
the rexx compiler returns the messages:
```
Error @ 15:54 - #INVALID_ASSEMBLER, "br c900"
Error @ 18:16 - #INVALID_ASSEMBLER, "br C100"
2 error(s) in source file
```

are branch statements disallowed?

**Comment by rvjansen** [comment]
It would be good if we have the missing loop statements - it will be a good investment for level C anyway

**Comment by rvjansen** [comment]
Peter, I  thought about this a bit, and I think that, for the runtime library, there is no reason we should not work in advance for a bit - we can use brexx, oorexx or regina to develop these functions in Rexx, and cut over the moment the support in the compiler is there. If we are lucky there is not a lot to change at that moment. Some things will depend on support of lower levels - those will have to wait.

**Comment by Peter-Jacob** [comment]
Hi René, agreed! I had the same idea, knocking something up with Regina, but at the moment I am not sure where to start. It's all about dependencies between bifs, maybe over several levels. Shall we add a new file in //.../rxas/bifname.dep which contains the dependencies (multi-level) of a single bif. Any other idea is appreciated.

**Comment by adesutherland** [comment]
The ASSEMBLER instruction is limited on purpose - it does not accept branches, labels and the like to ensure compiler generated code is correctly structured if you see what I mean.

Please add priority missing loops / rexx structures as issues - and I can prioritise the ones blocking progress :-)
